recipes/cuda-compiler/meta.yaml

```
@@ -0,0 +1,42 @@
+package:
+  name: cuda-compiler
+  version: 12.0.0
+
+source:
+  url: https://developer.download.nvidia.com/compute/cuda/redist/cuda_documentation/LICENSE.txt
+  sha256: 5db25d4fd138013b563f9a3d1d87f7de7df1dac014fd4cccdfbb3435a5cff761
+
+build:
+  number: 0
+  skip: true  # [osx or win]
+
+requirements:
+  run:
+    - c-compiler
+    - cxx-compiler
+    - cuda-cuobjdump 12.0.76
+    - cuda-cuxxfilt 12.0.76
+    - cuda-nvcc 12.0.76
+    - cuda-nvprune 12.0.76
+
+test:
+  commands:
+    - which cuobjdump
+    - which cu++filt
+    - which nvcc
+    - which nvprune
```

Maybe pre-emptively guard these, for when Windows support is added. I think we've done this kind of selector for tests on other packages where no Windows support is available yet.
```suggestion
    - which cuobjdump  # [linux]
    - which cu++filt   # [linux]
    - which nvcc       # [linux]
    - which nvprune    # [linux]
```